src/no-side-effects.ts

```diff
@@ -0,0 +1,62 @@
+// no-side-effects.ts
+
+/*
+ * Copyright (c) 2022-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import type { Rule } from 'eslint';
+
+const EXCLUDED_IDENTIFIERS = ['assert', 'debug', 'log'];
+
+export default {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Ensure no side effects can occur at the main module-level',
+      url: 'https://github.com/checkdigit/eslint-plugin',
+    },
+  },
+  create(context) {
+    return {
+      Program(node) {
+        const fileName = context.filename.split('src/')[1] ?? context.filename;
+        if (fileName.endsWith('spec.ts') || fileName.endsWith('test.ts') || fileName.endsWith('test.js')) {
+          return;
+        }
+        node.body.forEach((statement) => {
+          if (
+            (statement.type === 'ExpressionStatement' && statement.expression.type === 'AwaitExpression') ||
```

Comment:
this needs to be broken up somehow so its understandable.  a series of functions perhaps, not sure.  but this is too hard to parse.